pkgs/coverage/lib/src/formatter.dart

```diff
@@ -78,26 +80,27 @@ extension FileHitMapsFormatter on Map<String, HitMap> {
     String? basePath,
     List<String>? reportOn,
     Set<Glob>? ignoreGlobs,
+    bool Function(String path)? includeUncovered,
   }) {
     final pathFilter = _getPathFilter(
       reportOn: reportOn,
       ignoreGlobs: ignoreGlobs,
     );
     final buf = StringBuffer();
+
+    // Get all Dart files in the project
+    final allDartFiles = resolver.listAllDartFiles().toSet();
```

Comment:
Lots of duplicated code between `formatLcov` and `prettyPrint`. Can you factorize them?